gremlin-go/driver/performance-tests/java/PerformanceTests.java

```diff
@@ -0,0 +1,200 @@
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.__;
+import org.junit.Test;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.apache.tinkerpop.gremlin.process.traversal.AnonymousTraversalSource.traversal;
+
+class PerformanceTest {
+    final int SAMPLE_SIZE = 101;
+    final Cluster cluster = Cluster.build("localhost").port(8182).create();
+    final Client client = cluster.connect();
+    final Client g1client = client.alias("g");
+    final DriverRemoteConnection connection = DriverRemoteConnection.using(client, "ggrateful");
+    final GraphTraversalSource g = traversal().withRemote(connection);
+    final int REPEATS = 50;
+
+    public static GraphTraversal<?, ?> getTraversal(final String name, final GraphTraversalSource g) {
+        final Map<String, GraphTraversal<?, ?>> traversals = new HashMap<>();
+        traversals.put("getSongVertices", g.V().hasLabel("song"));
+        traversals.put("getNumVertices", g.V().count());
+        return traversals.get(name);
+
+    }
+
+    public static String[] getArgs(final int repeats) {
+        final String[] args = new String[repeats];
+        for (int i = 0; i < repeats; i++) {
+            args[i] = Integer.toString(i);
+        }
+        return args;
+    }
+
+    public static GraphTraversal<?, ?> getProjectTraversal(final GraphTraversalSource g, final int repeats, final String[] args) {
+        GraphTraversal<?, ?> traversal = g.V().project(args[0], Arrays.copyOfRange(args, 1, repeats));
+        for (int i = 0; i < repeats; i++) {
+            traversal = traversal.by(__.valueMap(true));
+        }
+        return traversal;
+    }
+
+    @Test
+    public void sample() {
+        System.out.println("~~~~~~~ RUNNING ONE ITEM PERFORMANCE TEST ~~~~~~~");
+        executeGetOneItemPerformanceTest();
+        System.out.println("~~~~~~~ RUNNING LIST PERFORMANCE TEST ~~~~~~~");
+        executeGetItemListPerformanceTest();
+        System.out.println("~~~~~~~ RUNNING TRANSFER PERFORMANCE TEST ~~~~~~~");
+        executeTransferPerformanceTest();
+        System.out.println("~~~~~~~ PERFORMANCE TESTS COMPLETE ~~~~~~~\"");
+    }
+
+    public timingData executeGetOneItemPerformanceTest() {
+        final List<Duration> durations = new ArrayList<>();
+        for (int i = 0; i < SAMPLE_SIZE; i++) {
+            try {
+                final String[] args = getArgs(REPEATS);
+                final Instant starts = Instant.now();
+                final Object temp = getProjectTraversal(g, REPEATS, args).next();
```

Comment:
What this variable is for? Never used?

Reply:
We are not using the object since we are only interested in the time it takes .next() to run

Reply:
code is updated to not store results in variable.